backend/src/main/java/ai/verta/modeldb/utils/RdbmsUtils.java

```diff
@@ -692,7 +693,11 @@ private static Predicate getOperatorPredicate(
    * @throws InvalidProtocolBufferException InvalidProtocolBufferException
    */
   public static Predicate getValuePredicate(
-      CriteriaBuilder builder, String fieldName, Path valueExpression, KeyValueQuery keyValueQuery)
+      CriteriaBuilder builder,
+      String fieldName,
+      Path valueExpression,
+      KeyValueQuery keyValueQuery,
+      boolean stringColumn)
```

Comment:
added flag to trim when we know the column is string, it is used when the predicate value is NUMBER